---
title: "<NuxtErrorBoundary>"
description: <NuxtErrorBoundary> 컴포넌트는 기본 슬롯에서 발생하는 클라이언트 사이드 오류를 처리합니다.
links:
  - label: 소스
    icon: i-simple-icons-github
    to: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/components/nuxt-error-boundary.vue
    size: xs
---

::tip
`<NuxtErrorBoundary>`는 내부적으로 Vue의 [`onErrorCaptured`](https://vuejs.org/api/composition-api-lifecycle.html#onerrorcaptured) 훅을 사용합니다.
::

## [이벤트](#events)

- `@error`: 컴포넌트의 기본 슬롯에서 오류가 발생했을 때 발생하는 이벤트입니다.

  ```vue
  <template>
    <NuxtErrorBoundary @error="logSomeError">
      <!-- ... -->
    </NuxtErrorBoundary>
  </template>
  ```

## [슬롯](#slots)

- `#error`: 오류가 발생했을 때 표시할 대체 콘텐츠를 지정합니다.

  ```vue
    <template>
      <NuxtErrorBoundary>
        <!-- ... -->
        <template #error="{ error, clearError }">
          <p>오류가 발생했습니다: {{ error }}</p>

          <button @click="clearError">오류 지우기</button>
        </template>
      </NuxtErrorBoundary>
    </template>
  ```

:read-more{to="/docs/getting-started/error-handling"}

## [예시](#examples)

### [스크립트에서 `error`와 `clearError`에 접근하기](#accessing-error-and-clearerror-in-script)

아래와 같이 컴포넌트의 스크립트 내에서 `error`와 `clearError` 속성에 접근할 수 있습니다:

```vue
<template>
  <NuxtErrorBoundary ref="errorBoundary">
    <!-- ... -->
  </NuxtErrorBoundary>
</template>

<script setup lang="ts">
const errorBoundary = useTemplateRef('errorBoundary')

// errorBoundary.value?.error
// errorBoundary.value?.clearError()
</script>
```
